metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java

```diff
@@ -1731,4 +1795,75 @@ public T getRequiredObjectProperty(T object, String name) {
      */
     public abstract void putSort(T object,String name,Sort s);
 
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // The above abstract methods were added based on need, and they sufffer badly from a strict top-down approach.
+    // There are multiple methods that do the same thing, and there are missing methods for crucial functionaloty.
+    // Don't use them. Use the ones below. 
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////    
```

Comment:
Should the above abstract methods be deprecated or have javadoc noting that they are not preferred? It might be useful to note what method is preferred for each.